Separate unit test statuses for WML exceptions and strict warnings

stevecotton opened the pull request:
This is part of working out whether a subset of the "fail" tests could be run in one Wesnoth instance. To run a test that returns TEST_FAIL_BROKE_STRICT with any other test would require a mechanism to reset lg::broke_strict()'s flag.

All tests that fail with an {ASSERT} will also set the lg::broke_strict() flag, the tests with the new status are only those that would pass without the strict flag.

In the SUF tests, change a test from fail-on-success to breaks-strict, rename the formula tests and add some comments. The rename is because "fai" is "Formula AI", an obsolete name for WFL.

CelticMinstrel commented:
This has nothing to do with your chances, but… what the heck does `formula="+max_moves"` actually test?

stevecotton commented:
It tests what happens when an invalid formulate is given to an SUF.

stevecotton commented:
The latest push rewrote the description, added comments and renamed the "fai" tests to "formula". The significant change is the description mentioning that all `{ASSERT}` failures also set the `lg:broke_strict()` flag.

stevecotton commented:
The AppVeyor build breakage may be caused by this PR, need to investigate further.

stevecotton commented:
Has a merge-conflict with #4603 , as they both change the run_wml_tests script. My plan is to get #4603 in first, and rebase this on top.
